rewrite-xml/src/main/java/org/openrewrite/xml/SemanticallyEqual.java

```diff
@@ -136,28 +139,42 @@ public Xml visitAttribute(Xml.Attribute attribute, Xml other) {
 
         @Override
         public Xml visitCharData(Xml.CharData charData, Xml other) {
-            if(charData == other) {
+            if (charData == other) {
                 return null;
             }
-            if(!(other instanceof Xml.CharData)) {
+            if (!(other instanceof Xml.CharData)) {
                 areEqual = false;
                 return null;
             }
-            Xml.CharData otherChar = (Xml.CharData)other;
-            if(!charData.getText().trim().equals(otherChar.getText().trim())) {
+            Xml.CharData otherChar = (Xml.CharData) other;
+            if (!charData.getText().trim().equals(otherChar.getText().trim())) {
                 areEqual = false;
                 return null;
             }
             return null;
         }
     }
+
     private static boolean isNullOrEmpty(@Nullable List<?> a) {
         return a == null || a.isEmpty();
     }
+
     private static boolean bothNullOrEmpty(@Nullable List<?> a, @Nullable List<?> b) {
         return isNullOrEmpty(a) && isNullOrEmpty(b);
     }
+
     private static boolean eitherNullOrEmpty(@Nullable List<?> a, @Nullable List<?> b) {
         return isNullOrEmpty(a) || isNullOrEmpty(b);
     }
+
+    private static boolean areTagsNamesEqual(Xml.Tag tag, Xml.Tag other, Cursor cursor) {
+        return tag.getLocalName().equals(other.getLocalName()) &&
+               tag.getNamespaceUri(cursor).equals(other.getNamespaceUri(cursor));
```

Comment:
This will only work if there a separate cursor instance for each tag. Any alternatives?